config.yaml

```diff
@@ -5,3 +5,12 @@ options:
   cluster-name:
     description: "Optional - Name of the MySQL InnoDB cluster"
     type: "string"
+# Config options for the legacy 'mysql relation'
+  user:
```

Comment:
We could consider naming these to reflect that their purpose is for a relation. Something like `mysql-interface-user` and `mysql-interface-database`?

Reply:
I like the idea of making the config name reflect the legacy relation interface. However, doing so will require consumer applications to make a code change (updating the config key [here](https://github.com/canonical/bundle-kubeflow/blob/master/releases/1.4/stable/kubeflow-lite/bundle.yaml#L60)) before being able to replace the `mariadb-k8s` charm for this charm

Reply:
Hmm. Interesting - @DomFleischmann how disruptive would this be?

Reply:
I think we might as well update to the new relation. It would still be interesting to see how this behaves during an update. What I expect us to do would basically be the following:

1. Deploy the new mysql charm alongside the new one.
2.  Upgrade the charms that consume it, for that release they will suppport both relations. 
3. Migrate the data to the new mysql
4. Relate the new mysql with the new charms
5. Delete the old relation and remove the old charms.

Does that sound like the right way of migrating to the new charms @shayancanonical ?

Reply:
> Do we have long lasting kubeflow deployments?

@shayancanonical Yes, and long lasting customer deployments by the time we move to the new charm.

> I believe that with this PR, one would be able to swap out mariadb-k8s with mysql-k8s without any difference (given that all tables have primary keys). This is what we call the legacy relation (support for which is introduced in this PR).

In that case if we change the config name it will only affect the new bundle with the new charm, this is not a big deal and can be done without an issue. 

The upgrade/migration path is something that probably requires more discussion in another forum.

Reply:
Changed config keys to include `mysql-interface-` prefix in 16a0a25a5b7c82698b3fa16c84266763cb9a4193